### DISCUSSION TITLE
SQL Editor - Format - Trim trailing spaces

### DISCUSSION THREAD

**Discussion by GuenterSch** [dbeaver/dbeaver#23459] [Q&A] [answered]
Hello,
to remove all trailing spaces of a script/file (opened in SQL Editor), it would be fine to have the functionality
**Edit>Format>Trim trailing spaces**
for the selection and for the whole file.

_Example:_
_ (Underscore) stands for a space
<CODE>select_\*________<P>
___from DUAL;<P>
--><P>
select\_\*<P>
___from DUAL;<P></CODE>

Thanks, BR
Günter

**Comment by LonwoLonwo** [comment]
Hello @GuenterSch 

![2022-05-31 23_17_07-DBeaver Ultimate 22 1 0 - _DBC_ Script-29](https://user-images.githubusercontent.com/45152336/171277372-9f740819-6642-4625-b7c6-c35df1bfd9bd.png)

You can highlight your statement `select_*________` and choose 'Trailing spaces'. And extra spaces will be trimmed.

**Comment by GuenterSch** [comment]
Hello Anastasiya,
thanks for your answer. I know already this funtionality to trim all spaces. But here also leading spaces will be removed, even though, that I want to have to indent the code.
**I need the funktionality to remove the trailing spaces (blanks) only.**
This shrinks the file and it improves a comparison between different versions of the "same" file.
BR
Günter

**Comment by LonwoLonwo** [comment]
Ok, thanks for the idea

**Comment by mstaks** [comment]
Hello Anastasiya,

It doesn't happen when I want to clear the spaces from the cursor to the first text on the right when aligning expressions.

this will be usefull to if you can do.  

Ctl+Alt+Delete   Delete from Cursor to the beginning of the line of text

similar but not like 

Ctl+Delete       Delete the next word

without delete next word just spaces, tabs

Thanks.

**Comment by alchemistmatt** [comment]
The latest update made Ctrl+Left Arrow and Ctrl+Right Arrow behave much more logically, by ignoring underscores (thank you!).  Since you're actively implementing text editor improvements, please also consider implementing this issue's request: to trim trailing whitespace (tabs and spaces). 

EditPad Pro has the option to [trim trailing whitespace](https://www.editpadpro.com/manual.html#extratrimauto) when you save a file, and I think that's a great way to implement this feature (besides, of course, the right click context menu to trim trailing whitespace immediately).

**Comment by TimonSP** [comment]
Visual Studio Code has the same feature "Files: Trim Trailing Whitespace" that "will trim trailing whitespace when saving a file"
It would be great help to have something similar in DBeaver.

**Comment by GuenterSch** [comment]
Hallo,
the idea of TimonSP is very good. Please consider, it would be necessary that we have two saves, one with trim and one without trim (current implementation).
Thanks, BR Günter

**Comment by anthonyvoya** [comment]
This seems like a trivial add that would be really helpful to those of us who only want to get rid of _trailing_ whitespace while leaving indents in place. In my experience it's also rather standard on modern editors.

**Comment by ShadelessFox** [comment, marked as answer]
We don't want to add new commands for such a simple task that can be accomplished using basic navigation shortcuts.

To remove leading spaces:
1. Press **Home**
2. Press **Ctrl+Backspace**

To remove trailing spaces:
1. Press **End**
2. Press **Ctrl+Delete**

**Reply by TimonSP** [reply]
The problem is to remove trailing spaces for ALL rows, do it for the WHOLE script.
How are your steps helping with this? Are you suggesting to go into EACH line and repeat described simple steps? 

Why you don't want to add such command? Or even setting for auto trim trailing spaces on file save event.

BTW, "Press End" move me to the end of all spaces of a line, so "Press Ctrl+Delete" removes new line instead of trailing spaces, so even for single row this approach doesn't work. Is it work differently on your side?

**Reply by ShadelessFox** [reply]
For some reason, I was thinking in the context of a single line. Doing this for a big script can be tedious indeed.

As a workaround, you can use the **Find/Replace** with the "Regular expressions" option enabled and replace all occurrences of `\s+$` with an empty string.

**Reply by ShadelessFox** [reply]
Since GitHub doesn't offer a way to convert a discussion back to its original ticket, I have created a new one: https://github.com/dbeaver/dbeaver/issues/23465

**Reply by anthonyvoya** [reply]
Agreed, thanks for creating a new one. This is a pretty standard and useful option on many IDEs.
